judge/forms.py

```diff
@@ -95,6 +97,28 @@ def __init__(self, *args, **kwargs):
         self.fields['user_script'].widget = AceWidget(mode='javascript', theme=user.profile.resolved_ace_theme)
 
 
+class EmailChangeForm(Form):
+    password = CharField(widget=forms.PasswordInput())
+    email = forms.EmailField()
+
+    def __init__(self, *args, user, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.user = user
+
+    def clean_email(self):
+        if User.objects.filter(email=self.cleaned_data['email']).exists():
+            raise ValidationError(_('This email address is already taken.'))
+        if is_email_address_bad(self.cleaned_data['email']):
+            raise ValidationError(_('Your email provider is not allowed due to history of abuse. '
+                                    'Please use a reputable email provider.'))
```

Comment:
The function name doesnt match the error message. Rename the function to something like `is_email_address_bad_domain` maybe.